### DISCUSSION TITLE
AirPlay

### DISCUSSION THREAD

**Discussion by Mohoganogan** [rern/rAudio#10] [Fixed Issues]
When install on a RaspberryPi 4 AirPlay will not turn on.  Via the web interface I can click the switch, it shows on.  However Airplay is not available to other devices. Upon returning to settings, the AirPlay switch is once again turned off.

**Comment by rern** [comment]
Fixed.
Please get today's update and try again.

**Comment by Mohoganogan** [comment]
Attempted to update via web interface. Received the following error.

Addons Progress


wget -q https://github.com/rern/rAudio-1/raw/main/install.sh
chmod 755 install.sh
./install.sh r1 Update main 
 !  Install file download failed.

Please try again.

Unable to log in via ssh using the username and password as specified in the documents.

Currently unable to update and test.

**Reply by klassikmann** [reply]
No problem to update here.

**Reply by Mohoganogan** [reply]
Suggestions on how you were able to update would be greatly appreciated.

**Reply by klassikmann** [reply]
Nothing special (no dirty tricks).

**Reply by Mohoganogan** [reply]
Are you able to use ssh to log in to rune audio?  I am told that the standard root/rune combination is not correct

**Reply by klassikmann** [reply]
root/ros

**Comment by cYCL157** [comment]
i updated to 20201221 via addons page without issue.  I was able to enable upnp, but airplay enables and then disables a few seconds later.  I did reboot after applying the update.

**Reply by cYCL157** [reply]
updated to 20201222 and still have the same issue with airplay switching back to off.

**Reply by rern** [reply]
Post your:
`cat /etc/shairport-sync.conf`
`systemctl status shairport-sync`

**Comment by Mohoganogan** [comment]
I am able to keep AirPlay on. 

Wifi turns itself off if a reboot/power down is performed.

Unable to rename the device under system-environment. Reverts to rAudio.

Installed via a fresh download and flash.

**Comment by Mohoganogan** [comment]
Started with fresh download
rAudio-1-RPi4.img.xz
results in system version 20201219

updated to 12-22 via web interface on RPI4

Lost mouse pointer on the RPI4 itself. Able to fumble around and click on things due to highlights of what is being pionted at

AirPlay will not enable/stay enabled. Status in lower left corner claims to be enabling, shows done, slider returns to off position. Not available for airplay from another device.

**Reply by rern** [reply]
Post your:
`cat /etc/shairport-sync.conf`
`systemctl status shairport-sync`

**Reply by Mohoganogan** [reply]
Updated to 20201223

rAudio:~  #  cat /etc/shairport-sync.conf
general = {
	name = "rAudio"
}
sessioncontrol = {
	run_this_before_play_begins = "/usr/bin/sudo /srv/http/bash/shairport.sh";
	run_this_after_play_ends = "/usr/bin/sudo /srv/http/bash/shairport.sh stop";
}
$alsa
alsa = {
	output_device = "hw:0";
	mixer_control_name = "HDMI";
}


rAudio:~  #  systemctl status shairport-sync
* shairport-sync.service - Shairport Sync - AirPlay Audio Receiver
     Loaded: loaded (/usr/lib/systemd/system/shairport-sync.service; enabled; vendor preset: disabled)
    Drop-In: /etc/systemd/system/shairport-sync.service.d
             `-override.conf
     Active: failed (Result: exit-code) since Wed 2020-12-23 13:22:10 UTC; 1min 17s ago
    Process: 329 ExecStart=/usr/bin/taskset -c 3 /usr/bin/shairport-sync (code=exited, status=1/FAILURE)
   Main PID: 329 (code=exited, status=1/FAILURE)

Dec 23 13:22:09 rAudio systemd[1]: Started Shairport Sync - AirPlay Audio Receiver.
Dec 23 13:22:10 rAudio shairport-sync[329]: fatal error: Line 8 of the configuration file "/etc/shairport-sync.conf":
                                            syntax error
Dec 23 13:22:10 rAudio systemd[1]: shairport-sync.service: Main process exited, code=exited, status=1/FAILURE
Dec 23 13:22:10 rAudio systemd[1]: shairport-sync.service: Failed with result 'exit-code'.


nano /etc/shairport-sync.conf

general = {
        name = "rAudio"
}
sessioncontrol = {
        run_this_before_play_begins = "/usr/bin/sudo /srv/http/bash/shairport.sh";
        run_this_after_play_ends = "/usr/bin/sudo /srv/http/bash/shairport.sh stop";
}
$alsa
alsa = {
        output_device = "hw:0";
        mixer_control_name = "HDMI";
}

**Comment by rern** [comment]
Run:
`sed -i '/$alsa/ d' /etc/shairport-sync.conf`
And enable AirPlay again.

**Reply by Mohoganogan** [reply]
I can now enable Airplay.

Observation though.  I have a monitor connected via HDMI. It has not sound built in.  System claims audio is being routed via head phone jack.  No sound out via jack. Remove monitor and reboot, audio is routed via audio jack.

**Reply by cYCL157** [reply]
this worked for me, thanks!

**Comment by ottl05** [comment]
I´ve got the same problem, 
Airplay won´t stay enabled. 
raudio1 on rpi4 (with todays update)
-- 
with: sed -i '/$alsa/ d' /etc/shairport-sync.conf
Airplay stays enabled
----

I´m coming from runeaudio re6 on a rpi1b and I want to use streaming from my iphone (bluetooth or Airplay). 

With bluetooth I´ve got no sound on my speakers (through receiver) On the playback page I see the songs played on my phone.
